contents/docs/session-replay/_snippets/ios-logging.mdx

```diff
@@ -0,0 +1,96 @@
+As console logs can contain sensitive information, we do not capture these logs automatically. 
+You can enable this feature from your client-side by setting `captureLogs = true` in our [iOS SDK config](/docs/session-replay/installation?tab=iOS#step-two-enable-session-recordings-in-your-project-settings).
+
+```ios_swift
+let config = PostHogConfig(apiKey: <ph_project_api_key>, host: <ph_client_api_host>)
+config.sessionReplay = true
+config.sessionReplayConfig.captureLogs = true // TIP: Needs to be configured in code. Project settings won't have an effect.
+
+PostHogSDK.shared.setup(config)
+```
+
+> **Important:** Individual console logs are truncated at 2000 characters. The rest of the log is not sent to PostHog. When truncating a log message we add ...[Truncated] to the message.
+
+### Custom log processing and sanitization
+
+PostHog iOS SDK allows you to customize how log messages are processed and categorized by providing a custom `processConsoleOutput` handler. 
+This handler receives raw console output and returns an array of processed log entries with their respective severity levels:
+
+```ios_swift
+let config = PostHogConfig(apiKey: <ph_project_api_key>, host: <ph_client_api_host>)
+
+// Custom log processing and sanitization
+config.sessionReplayConfig.captureLogsConfig.processConsoleOutput = { output in
```

Comment:
i feel like `processConsoleOutput` is the wrong name here cus if we change the way how we do things or we introduce `posthogCaptureLog(x)` method, the `consoleOutput` will be semantically wrong.
I suggest something else more generic like `logSanitizer`, `beforeLog`, etc, open to ideas

Reply:
> though one thing to clarify: should we split the buffer output by line and then call this handler for each line?

i think it really depends on how the buffer is structured, for example, if the user defines a regex match that requires the whole thing, it should be the whole thing.

NSLog('this is a log') -> this should trigger `logSanitizer` with 'this is a log'
NSLog('this is a log\nstill the same log') -> this should trigger `logSanitizer` with 'this is a log\nstill the same log'
so the call to `logSanitizer` should be based on how the log was called instead of just breaking it by line, if possible.

Reply:
I'd rather allow the user to drop or modify line, eg: `(String) -> String?`
and always use the returned line, that can be itself if not modified at all